contracts/test/LONStaking.t.sol

```diff
@@ -1156,6 +1159,95 @@ contract LONStakingTest is Test {
         lonStaking.permit(permit.owner, permit.spender, permit.value, permit.deadline, v, r, s);
     }
 
+    /*********************************
+     *         Test: conversion      *
+     *********************************/
+
+    function testEnableAndDisableConversion() public {
+        veLON veLon = new veLON(address(lon));
+
+        assertEq(address(lonStaking.veLon()), address(0x0));
+        assertEq(lonStaking.conversion(), false);
+
+        // enable the conversion
+        lonStaking.enableConversion(address(veLon));
+
+        assertEq(address(lonStaking.veLon()), address(veLon));
+        assertEq(lonStaking.conversion(), true);
+
+        // disable the conversion
+        lonStaking.disableConversion();
+
+        assertEq(address(lonStaking.veLon()), address(0x0));
+        assertEq(lonStaking.conversion(), false);
+    }
+
+    function _convertXLonToVeLonAndValidate(uint256 convertAmount, uint256 convertDuration) internal {
+        _stake(user, convertAmount);
```

Comment:
I think `stake()` should be handled before calling `_convertXLonToVeLonAndValidate`.